packages/containers/src/List/List.sagas.js

```diff
@@ -0,0 +1,76 @@
+import { put, takeEvery, take, call } from 'redux-saga/effects';
+import Connected from './List.connect';
+import Constants from './List.constant';
+import { getCollectionItems, sortList, filterList } from './selector';
+
+export function* updateList(data) {
+	const collectionId = data.props.collectionId || 'default';
+	const state = data.context.store.getState();
+	const componentState = state.cmf.components.getIn(['Container(List)', collectionId]);
+	const items = getCollectionItems(state, collectionId);
+
+	let results = items || data.props.config.items;
+	results = sortList(componentState, results, data.props.config);
+	results = filterList(componentState, results, data.props.config);
```

Comment:
Reassign a variable is hard to reason about. Plus the two selectors are not really selectors anymore, they do a holly bunch of thing, I don't even want to look inside x).

I suggest the following for the top part (here): 
```javascript
const stateItems = getCollectionItems(state, collectionId);
const propItems = data.props.config.items;

// Get data from state with simple selectors (sortBy, sortAsc, searchQuery, etc.)
// Doesn't even need a redux selector, just need to extract the lodash get

// Maybe extract before the starting List with checking if is an Immutable List if really needed.
const items = (stateItems || propItems || new List())
    .sort(sorting_fn)
    .filter(filtering_fn);
    // the sorting && filtering fn can be created with currying,
    // allowing config data injection reachable inside the final fn. 
```
In this way, the sorting and filter function doesn't deal with where the configs are, it's more portable and testable and more importantly readable :).